File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    static boolean[] visited;
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[n];
        
        for(int i= 0; i<n; i++){
            if(!visited[i]){
                 dfs(i, n, computers);
                answer++;
            }
        }
            
        return answer;
    }
    
    void dfs(int x, int n, int[][] computers){
        visited[x] = true;
        for(int i = 0; i<n; i++){
            if(computers[x][i] == 1 && !visited[i]){
                visited[i] = true;
                dfs(i, n, computers);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

bool visited[201] = {0,};

void dfs(int x, vector<vector<int>>& computers){
    visited[x] = true;
    for(int i = 0; i<computers.size(); i++){
        if(computers[x][i] == 1 && !visited[i]){
            dfs(i, computers);
        }
    }
}


int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    
    for(int i= 0; i<n; i++){
        if(!visited[i]) {
            dfs(i, computers); answer++;}
    }
    
    return answer;
}